timur/lib/client/jsx/components/query/query_use_query_clause.tsx

```diff
@@ -30,33 +38,40 @@ const useQueryClause = ({
   const attributeType = useMemo(() => {
     if ('' !== clause.attributeName) {
       const template = graph.template(clause.modelName);
-      if (!template) return 'text';
-
-      switch (
-        template.attributes[clause.attributeName].attribute_type.toLowerCase()
-      ) {
-        case 'string':
-          return 'text';
-        case 'date_time':
-          return 'date';
-        case 'integer':
-        case 'float':
-        case 'number':
-          return 'number';
-        case 'boolean':
-          return 'boolean';
-        case 'matrix':
-          return 'matrix';
-        default:
-          return 'text';
-      }
+      if (!template) return '';
+
+      return template.attributes[
+        clause.attributeName
+      ].attribute_type.toLowerCase();
     }
-    return 'text';
+    return '';
   }, [clause.attributeName, clause.modelName, graph]);
 
+  const fetchDistinctAttributeValues = useCallback(() => {
```

Comment:
My general concern here is cancellability.

Is it possible for fetchDistinctAttributeValues to be called multiple times before completion?  say a user pressing the button multiple times or navigating to and from the page?

Currently, it seems like this code would introduce a race condition in the above case -- that is, the order in which the .then operators get invoked after many calls to the original callback is non deterministic and likely to lead to confusion.

I have some tools in etna already to address this: `Cancellables` in `cancellable.jsx`.  But it's only a problem if the answer to #1 is yes.